conbench/metrics.py

```diff
@@ -192,7 +194,17 @@ def periodically_set_q_rem() -> None:
     def func():
         log.info("periodically_set_q_rem(): initiate")
         while True:
-            time.sleep(1)
+            # Delay for a bit until setting the gauge next time. But don't just
+            # sleep, to a busy sleep (responsive sleep loop that inspects
+            # SHUTDOWN often_.
+            deadline = time.monotonic() + 3
+            while time.monotonic() < deadline:
+                if job.SHUTDOWN:
```

Comment:
For the record: a module-level global is just a dictionary value, the key is the name. Threads can coordinates themselves through such simple data structures, i.e. this thread here immediate sees the update (boolean switch) when another thread runs code in job.py which does:

```
    global SHUTDOWN
    SHUTDOWN = True
```